src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1240,15 +1270,111 @@ private static MeshGeometry3D DrawTestMesh()
+
+            Camera.NearPlaneDistance = near;
+            Camera.FarPlaneDistance = far;
+        }
+
         /// <summary>
-        /// This method attempts to maximize the near clip plane in order to 
-        /// achiever higher z-buffer precision.
+        /// This method clamps the near and far clip planes around the scene geometry
+        /// to achiever higher z-buffer precision.
+        /// 
+        /// It does this by finding the distance from each GeometryModel3D object's corner points
+        /// to the camera plane. The camera's far clip plane is set to 2 * dfar, and the camera's 
+        /// near clip plane is set to nearPlaneDistanceFactor * dnear
         /// </summary>
-        internal void UpdateNearClipPlaneForSceneBounds(Rect3D sceneBounds)
+        internal static void ComputeClipPlaneDistances(Vector3 cameraPosition, Vector3 cameraLook, IEnumerable<Model3D> geometry, 
+            double nearPlaneDistanceFactor, out double near, out double far, double defaultNearClipDistance, double defaultFarClipDistance)
         {
-            // http: //www.sjbaker.org/steve/omniv/love_your_z_buffer.html
-            var maxDim = Math.Max(Math.Max(sceneBounds.SizeX, sceneBounds.Y), sceneBounds.SizeZ);
-            Camera.NearPlaneDistance = Math.Max(CalculateNearClipPlane(maxDim), 0.1);
+            near = defaultNearClipDistance; 
+            far = double.MaxValue;
+            
+            var bounds =
+                geometry.Where(i => i is GeometryModel3D)
+                    .Cast<GeometryModel3D>().Select(g=>g.Bounds());
+
+            // See http://mathworld.wolfram.com/Point-PlaneDistance.html
+            // The plane distance formula will return positive values for points on the same side of the plane
+            // as the plane's normal, and negative values for points on the opposite side of the plane. 
+
+            var distances = bounds.SelectMany(b => b.GetCorners()).
+                Select(c => c.DistanceToPlane(cameraPosition, cameraLook.Normalized())).
+                ToList();
+
+            distances.Sort();
+
+            if (distances.Any())
+            {
+                // All behind
+                // Set the near and far clip to their defaults
+                // because nothing is in front of the camera.
+                if (distances.All(d => d < 0))
+                {
+                    //Debug.WriteLine("All points behind.");
+                    near = defaultNearClipDistance;
+                    far = defaultFarClipDistance;
+                }
+                // All in front
+                // Set the near clip plane to some fraction of the 
+                // of the distance to the first point.
+                else if (distances.All(d => d > 0))
+                {
+                    //Debug.WriteLine("All points in front.");
+                    near = Math.Max(defaultNearClipDistance, distances.First(d => d > 0) * nearPlaneDistanceFactor);
```

Comment:
You are correct @ramramps. We can merge the last two cases. I hadn't realized late at night that they were the same. The `d=> d>0` then has to stay because the `distances` collection may have values `<0`.